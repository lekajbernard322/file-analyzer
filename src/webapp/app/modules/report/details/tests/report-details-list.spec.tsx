import React from "react";
import {render} from "@testing-library/react";
import ReportDetailsList from "../report-details-list";

const mockReport = {
    _id: "1",
    name: "report1",
    created: "some date",
    content: [
        {
            DLLName: "test.dll",
            Functions: ["function1", "function2"],
        },
    ],
}

const mockDllOccurrences = {
    "test.dll": 3,
};

describe("ReportDetailsList", () => {
    it("should render correctly", () => {
        const {container} = render(
            <ReportDetailsList report={mockReport} dllOccurrences={mockDllOccurrences} />
        );

        expect(container).toMatchSnapshot()
    })
})
