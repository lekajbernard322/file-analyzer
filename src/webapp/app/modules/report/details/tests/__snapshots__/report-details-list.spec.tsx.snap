// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDetailsList should render correctly 1`] = `
<div>
  <div>
    <ul
      class="report-details"
    >
      <li
        class="dll-name"
      >
        test.dll
         
        <span>
          (
          occurrences in other reports
          : 
          3
          )
        </span>
        <ul>
          <li
            class="function-name"
          >
            function1
          </li>
          <li
            class="function-name"
          >
            function2
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;
